core/admin/ExcAdminSite.py

```diff
@@ -0,0 +1,79 @@
+from django.contrib.admin import AdminSite
+from django.urls import NoReverseMatch, reverse
+from django.apps import apps
+from django.utils.text import capfirst
+
+
+class ExcursionAdmin(AdminSite):
+    index_template = "admin_index.html"
+
+    site_header = 'Excursion Admin'
+    site_title = 'Excursion Admin'
+    index_title = 'Admin Home'
+
+    def _build_app_dict(self, request, label=None):
+        """
+        Build the app dictionary. The optional `label` parameter filters models
+        of a specific app.
+        """
+        app_dict = {}
+
+        if label:
+            models = {
+                m: m_a for m, m_a in self._registry.items()
+                if m._meta.app_label == label
+            }
+        else:
+            models = self._registry
+
+        for model, model_admin in models.items():
+            app_label = model._meta.app_label
+
+            has_module_perms = model_admin.has_module_permission(request)
+            if not has_module_perms:
+                continue
+
+            perms = model_admin.get_model_perms(request)
+
+            # Check whether user has any perm for this module.
+            # If so, add the module to the model_list.
+            if True not in perms.values():
+                continue
+
+            info = (app_label, model._meta.model_name)
+            model_dict = {
+                'name': capfirst(model._meta.verbose_name_plural),
+                'object_name': model._meta.object_name,
+                'perms': perms,
+            }
+            # The changelist is hijacked in the viewable model admin to be the ViewList
+            if perms.get('view') or perms.get('change'):
+                try:
+                    model_dict['admin_url'] = reverse('admin:%s_%s_changelist' % info, current_app=self.name)
+                except NoReverseMatch:
+                    pass
```

Comment:
Do you have a plan for handling these exceptions? If not at least add an issue

Reply:
This is django's standard practice for this. If no admin is defined for a model, the admin should not be shown, but it should not be a lethal error